internal/app/splinter/api/api.go

```diff
@@ -0,0 +1,263 @@
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/gorilla/mux"
+	"github.com/mp-hl-2021/splinter/internal/app/splinter/model"
+	"log"
+	"net/http"
+	"reflect"
+	"strings"
+)
+
+type Api struct {
+	model.UserInterface
+}
+
+func NewApi(x model.UserInterface) Api {
+	return Api{x}
+}
+
+type Context struct {
+	*Api
+	wr  http.ResponseWriter
+	req *http.Request
+}
+
+type ErrorResponse struct {
+	Error      string
+	StatusCode int
+}
+
+func (ctx *Context) abortf(statusCode int, format string, a ...interface{}) {
+	ctx.wr.WriteHeader(statusCode)
+	t := ErrorResponse{
+		fmt.Errorf(format, a...).Error(),
+		statusCode,
+	}
+
+	if err := json.NewEncoder(ctx.wr).Encode(t); err != nil {
+		log.Fatal(err)
+	}
+
+	panic(t)
+}
+
+func (ctx *Context) abort(err error) {
+	ctx.abortf(http.StatusBadRequest, "%s", err.Error())
+}
+
+func (ctx *Context) token() model.Token {
+	header := ctx.req.Header.Get("Authorization")
```

Comment:
Please take a look at chat/api/api.go to implement the authorization functionality as middleware, cf. func (a *Api) authorize(handler http.HandlerFunc) http.HandlerFunc. Don't forget to return standard HTTP error codes.